Competitive Programming/c2cpp.md

```diff
@@ -0,0 +1,117 @@
+Most of the c++ fetures talked here work on c++98 and after but some work only after c++11 so for simplicity we will assume to be using c++11. Also we will be using gcc/g++ compiler. You can also use an IDE like codeblocks it already has gcc and g++ compilers by default (MINGW); remember to enable c++11 or c++14 option in codebloks' settings.
+
+Once you are familiar with C its easy to migrate to and learn C++. First of all since c++ is superset of C so you can use most of the concepts of c in c++. So we will first talk about porting code and then a couple of things new in c++ useful for competetive.
+
+# PREREQUISITES
+We require that you are atleast aware of the following concepts in c:
+
+* Basic Syntax of C (input, output, and logical, relational, arithmetic & bitwise operators):  
+* Loops
+* Arrays/Strings
+* Functions
+* Structures
+* Pointers
+
+If you don't know any of the above things, its better to have a recap first and than come back to this article.
+Also we WON'T require things like enums, unions, static, file IO etc. so as long as you are familiar with the ones above you are good to go.
+
+# PORTING C CODE
+So all of the things listed in prerequisites remain same in C++. You can use the same `scanf` and `printf` in C++ also; they are defined in header files <stdio.h> and <cstdio>. Similarly string functions are defined in header files <string.h> and <cstring>, math functions in math.h and <cmath>. Though the files listed in pairs are same for most intend and purposes but have a subtle difference which requires knowledge of namespaces. Though we will not go into this discussion but I have highlighted a few recources at the end of this section for any qurious mind. Though NOT recommended for beginners.
+
+For competitive programmers in general tend to use `#include<bits/stdc++.h>` which is basically a header file containing all the standard c++ header files (most STL files, header files for c etc.), though again its not a good developement practice to include unnecesarry headers but it is pretty common in competitive, as judges use runtime for time limit and this only increases my compile time and not my runtime. For more details refer to [geekforgeeks](http://www.geeksforgeeks.org/bitsstdc-h-c/)
+Also we will add another line to our code `using namespace std;` Reason is again related to namespaces but ignore this for now. If curious refer to the links at the end. Again NOT a reccommended read for beginners.
+
+In c++ we the return type of main() has to int.
+
+Keeping above 2 points in mind here is a sample program:
+
+```
+#include<bits/stdc++.h> //includes all header files
+using namespace std;
+
+int main()
+{
+int t;
+scanf("%d",&t);
+printf("%d",t);
+printf("Hello World")
+return 0;
+}
+```
+To compile this program on linux use `gcc --std=c++11 filename.cpp -o example` to run use `./example`. For codeblocks use build and run in build menu.
+
+# NEW in CPP (useful for competitive)
+## easier IO (cin and cout)
```

Comment:
Do you know about C++ IO being slower? I think it's worth mentioning that, since beginners will have no idea why they're getting TLE otherwise.